transcom/mymove
# B 18028 create sit delivery prime api end point

## B-18028

## Summary

This is the first part of a series of backlogs that will create a new Prime API endpoint to allow the prime to enter the customer contact date and the customer requested delivery date for shipments in Origin or Destination SIT. These values will eventually be used to automatically update the Authorized End Date for Origin and Destination SIT and Required Delivery Date for Origin SIT (B-17854 will add the calculations). This ticket is just to setup the initial API endpoint to allow the prime to enter the dates. 

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access the Prime Swagger UI page http://primelocal:3000/swagger-ui/prime.html#/mtoShipment/updateSITDeliveryRequest
2. Under the /mto-shipments/{mtoShipmentID}/sit-delivery section, click the Try it out button
4. Enter a MTO shipment ID and eTag for a shipment that is in ORIGIN or DESTINATION SIT. Move codes ORGSIT and DSTSIT are good ones to use if you don't have one that is already in SIT.
5. In the body section change the dates for sitCustomerContacted and sitRequestedDelivery
6. Click the Execute button
7. You should get a 200 success code with a response that shows the dates you entered (see Screenshots section for example)

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
![Screenshot 2023-12-11 at 12 51 25 PM](https://github.com/transcom/mymove/assets/149000222/8bc4d430-35d7-4148-aef3-fde61efb812e)
![Screenshot 2023-12-11 at 12 52 13 PM](https://github.com/transcom/mymove/assets/149000222/85aa8949-c709-4860-8859-1b07b333eab8)
